### PR TITLE
refactor(cli): cleanup compiler_snapshot and tsc/module_graph

### PR DESCRIPTION
**Pull request by kitsonk**
This refactors build and tsc on how they deal with "static" assets.  It drops about a 1mb out of the binary size and increases performance of TypeScript type checking by about 10% on my machine, and drops out about 300 LOCs.

Also refactored `tsc2` -> `tsc` and `module_graph2` -> `module_graph`.

### CODE REVIEW

File: cli/build.rs [hunk 1]
```diff
@@ -46,46 +50,139 @@ fn create_runtime_snapshot(snapshot_path: &Path, files: Vec<PathBuf>) {
...
+    "op_load",
+    json_op_sync(move |_state, args, _bufs| {
+      let v: LoadArgs = serde_json::from_value(args)?;
+      if v.specifier == build_specifier {
+        Ok(json!({
+          "data": r#"console.log("hello deno!");"#,
+          "hash": "1",
+          "scriptKind": 3
+        }))
+      } else if let Some(caps) = re_asset.captures(&v.specifier) {
```
Comment (bartlomieju) [inline cli/build.rs:160]: Could you add a comment what's going on here?

File: cli/build.rs [hunk 2]
```diff
@@ -46,46 +50,139 @@ fn create_runtime_snapshot(snapshot_path: &Path, files: Vec<PathBuf>) {
...
+          let path = if let Some(op_crate_lib) = op_crate_libs.get(lib) {
+            op_crate_lib.clone()
+          } else {
+            path_dts.join(format!("lib.{}.d.ts", lib))
+          };
+          let data = std::fs::read_to_string(path)?;
+          Ok(json!({
+            "data": data,
+            "hash": "1",
+            "scriptKind": 3
```
Comment (bartlomieju) [inline cli/build.rs:171]: Ditto

**Review by bartlomieju — APPROVED** [review]
LGTM with a minor nitpick

**Review by ry — APPROVED** [review]
Very nice work Kitson

**Pull request merged**